src/dbnode/persist/fs/retriever.go

```diff
@@ -434,18 +435,17 @@ func (r *blockRetriever) fetchBatch(
 	reqs := r.filterAndCompleteWideReqs(allReqs, seeker, seekerResources,
 		retrieverResources)
 
-	var limitErr error
-	if err := r.queryLimits.AnyExceeded(); err != nil {
-		for _, req := range reqs {
+	for _, req := range reqs {
```

Comment:
note: checking the cancel in this method is really just to cleanup the queue and prevent processing requests that will never be used. most likely the rpc routine has already checked the same deadline and is gone.